digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/LengthOfRoadAxisService.scala

```diff
@@ -0,0 +1,86 @@
+import fi.liikennevirasto.digiroad2.dao.linearasset.OracleLinearAssetDao
+import fi.liikennevirasto.digiroad2.linearasset.{NewLinearAsset, Value, LengthOfRoadAxisCreate, LengthOfRoadAxisUpdate}
+import fi.liikennevirasto.digiroad2.service.RoadLinkService
+import fi.liikennevirasto.digiroad2.util.PolygonTools
+import org.apache.commons.lang3.NotImplementedException
+
+
+
+class LengthOfRoadAxisService(roadLinkServiceImpl: RoadLinkService,
+                              eventBusImpl: DigiroadEventBus) extends DynamicLinearAssetService(roadLinkServiceImpl: RoadLinkService, eventBusImpl: DigiroadEventBus) {
+  override def roadLinkService: RoadLinkService = roadLinkServiceImpl
+
+  override def dao: OracleLinearAssetDao = new OracleLinearAssetDao(roadLinkServiceImpl.vvhClient, roadLinkServiceImpl)
+
+  override def municipalityDao: MunicipalityDao = new MunicipalityDao
+
+  override def eventBus: DigiroadEventBus = eventBusImpl
+
+  override def vvhClient: VVHClient = roadLinkServiceImpl.vvhClient
+
+  override def polygonTools: PolygonTools = new PolygonTools()
+
+  override def assetDao: OracleAssetDao = new OracleAssetDao
+
+  override def dynamicLinearAssetDao: DynamicLinearAssetDao = new DynamicLinearAssetDao
+  //crud
+  /// get
+  def getRaodwayLinear(): Unit = {
+
+
+    val id: Seq[Long] = Seq(0, 1)
+    id
+    throw new NotImplementedError
+  }
+
+  /// create
+  def createRoadwayLinear
+  (listOfAsset: List[LengthOfRoadAxisCreate],
+   username: String,
+   vvhTimeStamp: Long = vvhClient.roadLinkData.
+     createVVHTimeStamp()) = {
+    val addedElementId = withDynTransaction {
+      for (item <- listOfAsset) yield {
+        item.assetSequence.map{asset=>
+          val roadlink = roadLinkService.getRoadLinksAndComplementariesFromVVH(Set(asset.linkId), false).
+            find(_.linkId == asset.linkId)
+          super.createWithoutTransaction(item.typeId,asset.linkId ,asset.value,sideCode = 0,
+            measures = Measures(asset.startMeasure,asset.startMeasure),
+            username, vvhTimeStamp, roadlink,
+            verifiedBy = getVerifiedBy(username, item.typeId))
+        }
+      }
+    }
+    eventBusImpl.publish("LengthOfRoadAxisService:create",addedElementId.toSet)
+    addedElementId
+  }
+
+  def expireRoadwayLinear
+  (listOfAsset: List[LengthOfRoadAxisUpdate],
+   username: String
+  ) = {
+    val expiredElement=  withDynTransaction{
+      for (item <- listOfAsset) yield {super.expire(item.ids, username)}
+    }
+    eventBusImpl.publish("LengthOfRoadAxisService:expire",expiredElement.toSet)
+    expiredElement
+  }
+
+  /// Update
+  def updateRoadwayLinear
+  (listOfAsset: List[LengthOfRoadAxisUpdate],
+   username: String
+  ) = {
+   val modifiedElement=
+     withDynTransaction{
+       for (item <- listOfAsset) yield {super.updateWithoutTransaction(item.ids, item.value, username)}
+     }
+    eventBusImpl.publish("LengthOfRoadAxisService:update",modifiedElement.toSet)
+    modifiedElement
+  }
+}
```

Comment:
The idea was to use the current methods already existent on DynamicLinearAssetService, maybe we need to override the create, update and delete methods since on this user story it's needed to work with more than one asset at the same type.

But nothing too much, like receive multiple and then call the current methods multiples times

Reply:
In what way old one allows only one? If in frontend we create arrays of multiple assets of one type, we can run multiple post request to POST:/linearasset/ endpoint.  

Or was the idea that one roadlink can have multiple values. In one roadlink there might be K1 and K2 mark the same time for example.